libs/components/forms/src/lib/modules/file-attachment/file-attachment.component.ts

```diff
@@ -121,66 +141,99 @@ export class SkyFileAttachmentComponent
    * is complete.
    */
   @Input()
-  public required = false;
+  public required: boolean | undefined = false;
 
-  public set value(value: SkyFileItem) {
+  public set value(value: SkyFileItem | undefined) {
     // The null check is needed to address a bug in Angular 4.
     // writeValue is being called twice, first time with a phantom null value
     // See: https://github.com/angular/angular/issues/14988
     const isNewValue = value !== this.value && value !== null;
 
     if (isNewValue) {
-      this._value = value;
-      this._onChange(value);
-      this.updateFileAttachmentButton();
+      if (value) {
+        this.isImage = this.#fileItemService.isImage(value);
+      } else {
+        this.isImage = false;
+      }
+      this.#setFileName(value);
+
+      this.#_value = value;
+      this.#onChange(value);
+      this.#updateFileAttachmentButton();
     }
   }
 
-  public get value(): SkyFileItem {
-    return this._value;
+  public get value(): SkyFileItem | undefined {
+    return this.#_value;
   }
 
-  public currentThemeName: string;
+  public currentThemeName: string | undefined;
+
+  public fileName = '';
 
-  public showFileAttachmentButton: boolean;
+  public showFileAttachmentButton = true;
+
+  public truncatedFileName = '';
 
   @ViewChild('fileInput')
-  private inputEl: ElementRef;
+  public inputEl!: ElementRef;
 
   @ContentChildren(SkyFileAttachmentLabelComponent)
-  private labelComponents: QueryList<SkyFileAttachmentLabelComponent>;
+  public labelComponents:
+    | QueryList<SkyFileAttachmentLabelComponent>
+    | undefined;
+
+  public isImage = false;
+
+  #enterEventTarget: any;
+
+  #fileAttachmentId = uniqueId++;
+
+  #ngUnsubscribe = new Subject<void>();
 
-  private enterEventTarget: any;
+  #_disabled = false;
 
-  private fileAttachmentId = uniqueId++;
+  #_maxFileSize = 500000;
```

Comment:
I would save this as a `MAX_FILE_SIZE_DEFAULT` variable.

Reply:
Done